src/types/internal/schema.ts

```diff
@@ -221,15 +221,71 @@ export class Schema {
     return new Schema(names, fields, {refinement: Refinement.intersectionOf(schema1.refinement, schema2.refinement)});
   }
 
+  private static fieldTypeIntersection(field1: FieldType, field2: FieldType): FieldType|null {
+    const missingField1 = (field1 === null || field1 === undefined);
+    const missingField2 = (field2 === null || field2 === undefined);
+    if (missingField1 || missingField2) {
+      // TODO: Handle nullables
```

Comment:
Isn't returning null  already the desired behavior?

In any case, can we add bugs for this TODO and the one below?

Reply:
We need to handle nullables separately as both types `Foo {a: Int}` and `Foo {a: Int?}` match the type `Foo {a: Int?}`.

This will help us work towards Relaxed reads and writes where we want to be able to store types like
`Foo {a: Int}` and `Bar {b: Int}` together as `Foo? Bar? {a: Int?, b: Int?} [<a refinement that ties 'Bar-ness' to 'b' and 'Foo-ness' to 'a'`>]`

That refinement should allow us to query just for 'Foo's and get back a non-null 'a'. I don't think any of that can be done without handling nullables separately.

I'm happy to like the nullables and Relaxed reads and writes bugs though.

Reply:
Ah, got it!